Geometry/HcalCommonData/src/HcalDDDRecConstants.cc

```diff
@@ -600,7 +601,7 @@ void HcalDDDRecConstants::getOneEtaBin(int subdet, int ieta, int zside,
 				       std::map<int,int>& layers, bool planOne,
 				       std::vector<HcalDDDRecConstants::HcalEtaBin>& bins) const {
 
-  unsigned int lymax = (subdet == 1) ? 17 : 19;
+  unsigned int lymax = (subdet == static_cast<int>(HcalBarrel)) ? maxLayerHB_+1 : maxLayer_+1;
```

Comment:
@bsunanda - I'd suggest to convert enum HcalSubdetector to strongly typed enum class to make it safer and avoid extra temporaries and conversions.